src/google-maps/map-circle/map-circle.ts

```diff
@@ -0,0 +1,268 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="googlemaps" />
+
+import {Directive, Input, NgZone, OnDestroy, OnInit, Output} from '@angular/core';
+import {BehaviorSubject, combineLatest, Observable, Subject} from 'rxjs';
+import {map, take, takeUntil} from 'rxjs/operators';
+
+import {GoogleMap} from '../google-map/google-map';
+import {MapEventManager} from '../map-event-manager';
+
+/**
+ * Angular component that renders a Google Maps Circle via the Google Maps JavaScript API.
+ * @see developers.google.com/maps/documentation/javascript/reference/polygon#Circle
+ */
+@Directive({
+  selector: 'map-circle',
+})
+export class MapCircle implements OnInit, OnDestroy {
+  private _eventManager = new MapEventManager(this._ngZone);
+  private readonly _options = new BehaviorSubject<google.maps.CircleOptions>({});
+  private readonly _center =
+      new BehaviorSubject<google.maps.LatLng|google.maps.LatLngLiteral|undefined>(undefined);
+  private readonly _radius = new BehaviorSubject<number|undefined>(undefined);
+
+  private readonly _destroyed = new Subject<void>();
+
+  /**
+   * Underlying google.maps.Circle object.
+   *
+   * @see developers.google.com/maps/documentation/javascript/reference/polygon#Circle
+   */
+  circle: google.maps.Circle;  // initialized in ngOnInit
+
+  @Input()
+  set options(options: google.maps.CircleOptions) {
+    this._options.next(options || {});
+  }
+
+  @Input()
+  set center(center: google.maps.LatLng|google.maps.LatLngLiteral) {
+    this._center.next(center);
+  }
+
+  @Input()
+  set radius(radius: number) {
+    this._radius.next(radius);
+  }
+
+  /**
+   * @see
+   * developers.google.com/maps/documentation/javascript/reference/polygon#Circle.center_changed
+   */
+  @Output()
+  centerChanged: Observable<void> = this._eventManager.getLazyEmitter<void>('center_changed');
```

Comment:
`circleCenterChanged`?

Reply:
In the past we've used a component prefix for certain events because they were mouse events that had non-standard behavior (For example, the Polyline had specialized PolyMouseEvents), but we haven't been using them for events that are specific to those components.